server/conn.go

```diff
@@ -1082,6 +1082,19 @@ func (cc *clientConn) Run(ctx context.Context) {
 			disconnectByClientWithError.Inc()
 			return
 		}
+		// get max allowed packet length and compare
```

Comment:
Could you add a test?

Reply:
I tried to add an IT in `explaintest` in commit [102032d](https://github.com/pingcap/tidb/pull/32216/commits/102032d41175758b7454fa0a7a6462c1257b0d20), but it failed in test `new_character_set`. So I revert it.